src/tests/ef_vi/eflatency.c

```diff
@@ -214,24 +222,40 @@ generic_ping(struct eflatency_vi* rx_vi, struct eflatency_vi* tx_vi,
 {
   struct timeval start, end;
   int i;
-  int do_rx_post = ( rx_vi->vi.nic_type.arch != EF_VI_ARCH_EFCT );
+  const int do_rx_post = ( rx_vi->vi.nic_type.arch != EF_VI_ARCH_EFCT );
+  uint8_t * const tx_pkt = pkt_bufs[FIRST_TX_BUF]->dma_buf;
+  uint8_t pattern = 0;
+
+  for( i = 0; i < cfg_warmups; ++i, pattern++ ) {
+    memset(&tx_pkt[HEADER_SIZE], pattern, cfg_payload_len);
+    checksum_udp_pkt(tx_pkt);
 
-  for( i = 0; i < cfg_warmups; ++i ) {
     tx_send(tx_vi);
     if( do_rx_post )
       rx_post(&rx_vi->vi);
     rx_wait(rx_vi);
-   generic_desc_check(tx_vi, 0);
+    generic_desc_check(tx_vi, 0);
   }
 
   gettimeofday(&start, NULL);
 
-  for( i = 0; i < cfg_iter; ++i ) {
+  for( i = 0; i < cfg_iter; ++i, pattern++ ) {
+    memset(&tx_pkt[HEADER_SIZE], pattern, cfg_payload_len);
+    checksum_udp_pkt(tx_pkt);
```

Comment:
The `memset()` and `checksum_udp_pkt()` calls are outside of the timing loop for the `ping` process, so it shouldn't affect the measurement results.  It might add a minimal amount of overhead to the send, although since there are also `printf()` and other calls that the process does as part of reporting the results, I'm not sure if you could even tell...
On the receive side (either `pong` or the `ping` process receiving the reply), it seems to add about 150ns per side on my test machine.